src/sql/upgrade/update_db_lutece_core-5.1.0-5.2.0.sql

```diff
@@ -0,0 +1,21 @@
+ALTER TABLE core_admin_user MODIFY COLUMN password LONG VARCHAR default NULL;
+ALTER TABLE core_user_password_history COLUMN MODIFY password LONG VARCHAR NOT NULL;
+
+-- update password storage
+UPDATE core_admin_user SET password =
+CONCAT(COALESCE(
+	(SELECT CONCAT(cd1.entity_value,':') FROM core_datastore cd1 CROSS JOIN core_datastore cd2	WHERE cd1.entity_key = 'core.advanced_parameters.encryption_algorithm' AND cd2.entity_key = 'core.advanced_parameters.enable_password_encryption' AND LOWER(cd2.entity_value) = 'true'),
+	'PLAINTEXT:')
+,password);
+
+-- updating password history with best effort to guess format
+-- for PostgreSQL, replace 'REGEXP' by '~*' and 'NOT REGEXP' by '!~*'
+UPDATE core_user_password_history SET password = CONCAT('MD5:',password) WHERE password REGEXP '^[0-9a-f]{32}$';
+UPDATE core_user_password_history SET password = CONCAT('SHA-1:',password) WHERE password REGEXP '^[0-9a-f]{40}$';
+UPDATE core_user_password_history SET password = CONCAT('SHA-256:',password) WHERE password REGEXP '^[0-9a-f]{64}$';
+UPDATE core_user_password_history SET password = CONCAT('PLAINTEXT:',password) WHERE password NOT REGEXP '^(MD5:[0-9a-f]{32}|SHA-1:[0-9a-f]{40}|SHA-256:[0-9a-f]{64})$'
```

Comment:
I think we should truncate these tables instead, so that all unsafe passwords are erased by default. I think that this table is only used to prevent choosing the same password again, so it's not very important if we delete everything.

What do you think ?